requests-for-comments/2024-01-23-project-environments.md

```diff
@@ -0,0 +1,189 @@
+
+	They also noted it's annoying how switching projects from a dashboard takes you back to the homepage. Sadly, this is currently the only thing we can do, because a dashboard is strictly accessible in only one project.
+
+- [Sharing event definitions - P ZEN-5939](https://posthoghelp.zendesk.com/agent/tickets/5939) and [ZEN-5937](https://posthoghelp.zendesk.com/agent/tickets/5937) (two from one org)
+
+	They don't want event and property definitions to get out of sync between staging and production projects, since they intend the shape of data to be the same for all environments. 
+
+- [Sharing event definitions - I ZEN-4637](https://posthoghelp.zendesk.com/agent/tickets/4637)
+
+	Same as above, with a focus on event descriptions.
+
+- [Experiment copying - S ZEN-8439](https://posthoghelp.zendesk.com/agent/tickets/8439)
+
+	They would like to copy experiments between staging and production. This is framed as "would be nice" though.
+
+- [Feature flag copying - O ZEN-10377](https://posthoghelp.zendesk.com/agent/tickets/10377)
+
+	Wanted to copy flags from dev to production.
+
+- [Feature flag copying - K ZEN-10652](https://posthoghelp.zendesk.com/agent/tickets/10652)
+
+	Same as above.
+
+#### Issues
+
+- [Using projects as environments: syncing actions/insights/dashboards/playlists etc. #13418](https://github.com/PostHog/posthog/issues/13418)
+
+	5 users have chimed in to express their need for a better UX around environments, and 11 more have upvoted a comment suggesting handling of "environments". All commenters mention insights and dashboards as a problem. Feature flags and event/property definitions are also mentioned.
+
+- [Copy / Move dashboards across projects #19476](https://github.com/PostHog/posthog/issues/19476)
+
+	3 users commented to express their need, one specifically mentioned dashboards, insights, and actions as currently problematic.
+
+#### User interviews
+
+- [Feature flag & experiment & dashboard copying - R, by Annika](https://posthog.slack.com/archives/C011L071P8U/p1671640760520999)
+
+	They run development and staging, same as customers above they find manual replication tedious and error-prone.
+
+### Summarizing needs
+
+It looks like **users always need to keep the data isolated** between environments – features: *events*, *people*, *groups*, *recordings*. *Toolbar authorized URLs* should also likely be different, due to separate domains.
+
+**Analysis setup should be shared** – crucially features: *dashboards*, *insights*, *actions*, and *cohorts*. Often also: *event and property definitions*, and *group types*. Shared *feature flags* are also a common ask, with *experiments* sometimes mentioned.
+
+I've never heard of the following features mentioned in terms of environments: *playlists*, *surveys*, *early access features*, *annotations*, *data warehouse tables*, *apps*, or *batch exports*. Playlists seem similar enough to dashboards that they should also be shared. As for the rest, it's hard to say.
```

Comment:
Apps are indeed a tricky situation, I see the value in having them synced for consistency, but I feel like per-environment makes the most sense:
  - as a user, I'd love to test out transforms on the dev dataset before promoting them to the prod dataset
  - exports will probably be by environment, as you don't need to archive dev data, or you'll want it in a different warehousing dataset. We could achieve this by filtering the events by environment, but it feels wasteful

It would be relatively easy to have plugin-server support a mix of per-project and per-environment pluginconfigs, but the UI for this might get confusing?

Reply:
Good points, I will make it explicit that apps will be environment-specific